config/src/main/java/org/springframework/security/config/annotation/web/configurers/oauth2/client/OAuth2ClientConfigurer.java

```diff
@@ -218,6 +218,10 @@ private void configure(B builder) {
 			OAuth2AuthorizationRequestRedirectFilter authorizationRequestRedirectFilter = createAuthorizationRequestRedirectFilter(builder);
 			builder.addFilter(postProcess(authorizationRequestRedirectFilter));
 			OAuth2AuthorizationCodeGrantFilter authorizationCodeGrantFilter = createAuthorizationCodeGrantFilter(builder);
+			RequestCache requestCache = builder.getSharedObject(RequestCache.class);
+			if (requestCache != null) {
+				authorizationCodeGrantFilter.setRequestCache(requestCache);
```

Comment:
Please add a test in `OAuth2ClientConfigurerTests` to `verify(requestCache).getRequest()`